### PR TITLE
Last-rite more extraneous Haskell packages

### PR DESCRIPTION
**Pull request by hololeap**
I found quite a few more packages that are not needed to build the "core leaf" Haskell packages that will remain on `::gentoo`. This is assuming our core group consists of:

```
app-admin/haskell-updater
app-portage/hackport
app-text/pandoc
dev-util/shellcheck
dev-vcs/darcs
dev-vcs/git-annex
x11-misc/xmobar
x11-wm/xmonad
x11-wm/xmonad-contrib
```

Some more packages were found by grepping `::gentoo` for the string `"haskell-cabal"`.

A commit will also be needed in `::haskell` ala https://github.com/gentoo-haskell/gentoo-haskell/commit/f30501d9c93f8a44bdbadbcc48aa5b932b739666

---

Edit:

I [now realize](https://github.com/gentoo/gentoo/pull/26837#issuecomment-1213361818) that `dev-haskell/cabal-install` should be added to the "core-group".

### CODE REVIEW

File: profiles/package.mask [hunk 1]
```diff
@@ -33,6 +33,137 @@
...
+dev-haskell/filemanip
+dev-haskell/filepattern
+dev-haskell/fsnotify
+dev-haskell/ghc-lib-parser
+dev-haskell/ghc-lib-parser-ex
+dev-haskell/githash
+dev-haskell/groupoids
+dev-haskell/groups
+dev-haskell/hackage-security
+dev-haskell/haddock
```
Comment (hololeap) [inline profiles/package.mask:86]: Surprisingly, the `haddock` executable is not needed for building documentation for Haskell packages. The built-in `haddock` command of Cabal is used internally instead.

**Comment by hololeap** [comment]
It looks like I forgot to include a package in the "core group":

```
dev-haskell/cabal-install
```

File: profiles/package.mask [hunk 2]
```diff
@@ -33,6 +33,134 @@
...
+# are available in ::haskell.
+# Removal on 2022-09-12.
+app-portage/fquery
+dev-games/goatee
+dev-haskell/megaparsec
+dev-haskell/parser-combinators
+dev-haskell/megaparsec-tests
+dev-haskell/hspec-megaparsec
+dev-haskell/neat-interpolation
+dev-haskell/stack
```
Comment (solpeth) [inline profiles/package.mask:51]: Users have been complaining about stack not working in the bug tracker. Is it worth keeping? I personally don't see why but some upstreams use it for development.

File: profiles/package.mask [hunk 3]
```diff
@@ -33,6 +33,134 @@
...
+dev-haskell/haddock-api
+dev-haskell/hashable-time
+dev-haskell/hashtables
+dev-haskell/haskell-src-meta
+dev-haskell/hdbc
+dev-haskell/hdbc-odbc
+dev-haskell/hdbc-postgresql
+dev-haskell/hdbc-sqlite3
+dev-haskell/hi-file-parser
+dev-haskell/hlint
```
Comment (solpeth) [inline profiles/package.mask:94]: I suspect we'll get a bug to bring this back, but until then I support its removal.

**Review by solpeth — APPROVED** [review]
Looks good!

**Comment by l29ah** [comment]
/me petitions x11-misc/xmonad-extras for the core group

File: profiles/package.mask (hunk 3 above)
Comment (hololeap) [reply profiles/package.mask:94]: `hlint` is really only for people developing in Haskell, so it makes sense to me to only have it in `::haskell` (same with `stack`).

**Comment by hololeap** [comment]
@l29ah perhaps `xmonad-screenshot`, `xmonad-utils`, and `xmonad-wallpaper` as well?

**Comment by hololeap** [comment]
I'm closing https://github.com/gentoo/gentoo/pull/26835 and merging those changes here, because it is getting annoying to maintain the two branches.

**Comment by thesamesam** [comment]
>In order to force reassignment and/or bug reference scan, please append [please reassign] to the pull request title.

Could you do this please? IIRC some of these packages aren't maintained by haskell@.

**Comment by matoro** [comment]
> @l29ah perhaps `xmonad-screenshot`, `xmonad-utils`, and `xmonad-wallpaper` as well?

These aren't currently in ::gentoo, same for xmonad-extras, are we adding them or what?

**Comment by hololeap** [comment]
@matoro I'm not sure. In theory, it seems to me that they should be in ::gentoo if `xmonad` is, but I'm not sure how many extra dependencies they will pull in. I'll do a bit of testing.

**Comment by hololeap** [comment]
@thesamesam I've tested compiling all the core packages (and dependencies) in my sandbox, and I don't see any issues. This should be ready to pull.

@l29ah We can add `xmonad-extras` in a later PR, but I want to get this out the door.

**Pull request closed without merging**